### DISCUSSION TITLE
Wiecej opcji oznaczania linijek w zakładce Osobiste

### DISCUSSION THREAD

**Discussion by UniqueTM2** [TomaszGasior/RadioLista-v3#163] [2. Pomysły Ideas]
Mam pytanie, czy dałoby radę wprowadzić więcej opcji oznaczania danej linijki? Obecnie można wybrać 3 kolory, pogrubić, albo przekreślić. Według mnie przydałoby się więcej kolorów i np. użycie dwóch opcji, czyli ustawić kolor i pogrubić, albo pogrubić i przekreślić. Czyli taka opcja wielokrotnego wyboru.
Czasem potrzebuje oznaczyć wyłaczony nadajnik przekreśleniem, ale ze to stacja zagraniczna, którą oznaczam pogrubieniem, to przekreślajac nie ma pogrubienia.

**Comment by TomaszGasior** [comment]
Witaj na forum. :) Możesz jeszcze zasubskrybować repozytorium przyciskiem „Watch” u góry po prawej, żeby dostawać powiadomienia o ogłoszeniach.

Na pierwszy rzut oka wydaje mi się, że nie ma takiej potrzeby, aby mnożyć możliwości kolorowania i personalizowania wierszy wykazu — fakt, że stacja jest zagraniczna można odczytać z pola „Kraj”. Z drugiej strony widzę, że korzystasz z serwisu  aktywnie, prowadzisz bardzo bogaty wykaz i spróbuję zdać się na twoje doświadczenie. :)

 Obecnie rzeczywiście można zaznaczyć tylko jedną opcję personalizacji wyglądu wiersza. Pochodzi to jeszcze z czasów, gdy w RadioLiście były motywy (#148)  i było to konieczne do zapewnienia odpowiedniego kontrastu tekstu w każdym z nich (co i tak wcale się nie udawało często…).

Z twojej prośby wnioskuję, że z opcji jednokrotnego wyboru musielibyśmy przejść do formy, gdzie użytkownik może w całkowitej separacji włączyć i wyłączyć następujące opcje:
- kolor tła wiersza: brak | czerwony | zielony | niebieski | żółty | różowy | brązowy | etc.
- pogrubienie
- pochylenie
- przekreślenie

Nie chciałbym dawać możliwości dowolnego wyboru koloru, a pozostać przy predefiniowanych opcjach, aby być w stanie zapewnić dobry kontrast tekstu.

Ze względu na dotychczasową już istniejącą opcję wyróżnienia wizualnego, wprowadzenie nowej opcji będzie wymagało migracji ustawień ze starej opcji do nowej — nie jest to niemożliwe :), jedynie zaznaczam to dla specyfikacji zadania.

Nowa opcja będzie też wymagała innej prezentacji na ekranie dodawania/edycji stacji, ze względu na więcej jej wariantów. Być może zaimplementuję ją finalnie dopiero po wprowadzeniu nowego designu całego serwisu (#139). Niestety, ze względów prywatnych moja praca w temacie nowego designu stanęła w miejscu i ruszy dopiero wkrótce.

Nie mogę więc obiecać konkretnego terminu, ale zaraz dodam i przypiszę sobie odpowiednie zadanie, które będziesz mógł również zasubstrybować, żeby być na bieżąco z postępem.

**Reply by UniqueTM2** [reply]
Jest kraj, zgadzam sie, ale pogrubienie stacji zagranicznej w moim przypadku pomaga, wizualnie jest tak dla mnie lepiej. Co do kolorów to tak w sumie zaproponowałem, ale sa trzy i może tak zostać. Separacja opcji jednak uważam, że jest potrzebna. Każdy z nas tworzy wykaz inaczej, jedni oznaczaja kolorem rodzaj propagacji, ale np. tez chca pogrubic, aby z jakichś powodów wyróżnić dana stację. Nie ukrywam, opcja niezależnego użycia tych opcji bardzo pomogłaby mi w pracy z wykazem, bo aktywnośc we wprowadzaniu to jedno, ale ja mam ten główny wykaz otwarty zawsze jak skanuje, a jestem typowym wzrokowcem :). 
Dziekuje z góry Tomku za zainteresowanie sie tematem, jeśli uda sie zaimplementowac do opcji moje sugestie, będę bardzo, bardzo wdzięczny.
Pozdro

**Reply by TomaszGasior** [reply]
Stworzyłem zadanie, które możesz zasubkrybować, żeby widzieć postęp: https://github.com/TomaszGasior/RadioLista-v3/issues/164

Postaram się ruszyć ten temat w tym miesiącu.

**Comment by TomaszGasior** [comment]
Zmiana została wykonana i wdrożę ją na radiolista.pl jutro. :) https://github.com/TomaszGasior/RadioLista-v3/discussions/173
